tracee-ebpf/tracee/argprinters.go

```diff
@@ -51,130 +51,197 @@ func (t *Tracee) parseArgs(event *external.Event) error {
 		if int32(event.EventID) == CapCapableEventID {
 			if capArg := getEventArg(event, "cap"); capArg != nil {
 				if capability, isInt32 := capArg.Value.(int32); isInt32 {
-					capArg.Value = helpers.ParseCapability(capability)
 					capArg.Type = "string"
+					capArg.Value = ""
+					capabilityFlagArgument, err := helpers.ParseCapability(uint64(capability))
+					if err == nil {
+						capArg.Value = capabilityFlagArgument.String()
+					}
 				}
 			}
 		}
 		if int32(event.EventID) == SecurityFileOpenEventID {
 			if flagsArg := getEventArg(event, "flags"); flagsArg != nil {
 				if flags, isInt32 := flagsArg.Value.(int32); isInt32 {
-					flagsArg.Value = helpers.ParseOpenFlags(uint32(flags))
 					flagsArg.Type = "string"
+					flagsArg.Value = ""
+					openFlagArgument, err := helpers.ParseOpenFlagArgument(uint64(flags))
+					if err == nil {
+						flagsArg.Value = openFlagArgument.String()
+					}
 				}
 			}
 		}
 	case MmapEventID, MprotectEventID, PkeyMprotectEventID:
 		if protArg := getEventArg(event, "prot"); protArg != nil {
 			if prot, isInt32 := protArg.Value.(int32); isInt32 {
-				protArg.Value = helpers.ParseMemProt(uint32(prot))
+				mmapProtArgument := helpers.ParseMmapProt(uint64(prot))
+				protArg.Value = mmapProtArgument.String()
 				protArg.Type = "string"
 			}
 		}
 	case PtraceEventID:
 		if reqArg := getEventArg(event, "request"); reqArg != nil {
 			if req, isInt64 := reqArg.Value.(int64); isInt64 {
-				reqArg.Value = helpers.ParsePtraceRequest(req)
 				reqArg.Type = "string"
+				reqArg.Value = ""
+				ptraceRequestArgument, err := helpers.ParsePtraceRequestArgument(uint64(req))
+				if err == nil {
+					reqArg.Value = ptraceRequestArgument.String()
+				} // TODO: https://github.com/aquasecurity/libbpfgo/issues/108
 			}
 		}
 	case PrctlEventID:
 		if optArg := getEventArg(event, "option"); optArg != nil {
 			if opt, isInt32 := optArg.Value.(int32); isInt32 {
-				optArg.Value = helpers.ParsePrctlOption(opt)
 				optArg.Type = "string"
+				optArg.Value = ""
+				prctlOptionArgument, err := helpers.ParsePrctlOption(uint64(opt))
+				if err == nil {
+					optArg.Value = prctlOptionArgument.String()
+				}
 			}
 		}
 	case SocketEventID:
 		if domArg := getEventArg(event, "domain"); domArg != nil {
 			if dom, isInt32 := domArg.Value.(int32); isInt32 {
-				domArg.Value = helpers.ParseSocketDomain(uint32(dom))
 				domArg.Type = "string"
+				domArg.Value = ""
+				socketDomainArgument, err := helpers.ParseSocketDomainArgument(uint64(dom))
+				if err == nil {
+					domArg.Value = socketDomainArgument.String()
+				}
 			}
 		}
 		if typeArg := getEventArg(event, "type"); typeArg != nil {
 			if typ, isInt32 := typeArg.Value.(int32); isInt32 {
-				typeArg.Value = helpers.ParseSocketType(uint32(typ))
 				typeArg.Type = "string"
+				typeArg.Value = ""
+				socketTypeArgument, err := helpers.ParseSocketType(uint64(typ))
+				if err == nil {
+					typeArg.Value = socketTypeArgument.String()
```

Comment:
About the "String"-ification discussion. Independently of what is right, IMHO, this follows recent changes to libbpfgo (about having a specific Type and a String() to it, so we can control the type's values and do 1:1 with real ones).